src/pages/add-item/components/AddItemForm.jsx

```diff
@@ -0,0 +1,190 @@
+import Input from './Input';
+import { useState, useRef, useEffect } from 'react';
+import TagInput from './TagInput';
+
+const AddItemForm = () => {
+  const [formData, setFormData] = useState({
+    name: '',
+    description: '',
+    price: '',
+    tag: [],
+    img: null,
+  });
+
+  const [tagInput, setTagInput] = useState('');
+  const [productImg, setProductImg] = useState(null);
+  const [previewImg, setPreviewImg] = useState(null);
+  const [isSubmitable, setIsSubmitable] = useState(false);
+  const fileInputRef = useRef(null);
+
+  useEffect(() => {
+    const { name, description, price, tag } = formData;
+    const allFilled =
+      name.trim() &&
+      description.trim() &&
+      price.trim() &&
+      Array.isArray(tag) &&
+      tag.length > 0;
+    setIsSubmitable(!!allFilled);
+  }, [formData]);
```

Comment:
💊 제안
form이 valid할 경우를 판단하는 isSubmitable state의 경우 formData state로 대체가 가능할 것 같아요.

```suggestion
  const fileInputRef = useRef(null);

  const isFormValid = () => {
    const { name, description, price, tag } = formData;
    
    return  name.trim() &&
      description.trim() &&
      price.trim() &&
      Array.isArray(tag) &&
      tag.length > 0;
  }
  
  // 사용시 : <button disabled={!isFormValid()}>등록</button>
```